# Approximate Confidence Intervals (cont.)

Recall from a recent CBS News/YouGov poll (https://www.cbsnews.com/news/poll-trump-leads-biden-economy/), that from a sample of $n = 2159$ adults, $\hat p = 0.48$ would vote for current president Joe Biden over former president Donald Trump in the upcoming presidential election.

Let's use this data to compute both a Wald and a Wilson confidence interval.

We'll start with Wald.  For a given confidence level $1 - \alpha$, the endpoints of the Wald interval are

$$\hat p \pm z_{\alpha/2}\sqrt{\frac{\hat p(1 - \hat p)}{n}}$$

Recall that $z_{\alpha/2}$ is the solution to $P(Z > z_{\alpha/2}) = \frac{\alpha}{2}$, where $Z\sim N(0,1)$.

In [2]:
z <- qnorm(0.025, lower.tail=FALSE)
n <- 2159
pHat <- 0.48
E <- z*sqrt(pHat*(1 - pHat)/n)
E
pHat - E
pHat + E

[1] 0.02107386

[1] 0.4589261

[1] 0.5010739

So we're 95% confident that the population proportion of Biden voters is $p\in(0.45893, 0.50107)$.

Now let's switch to the Wilson interval.  Recall that this is

$$\frac{1}{1 + \frac{z_{\alpha/2}^2}{n}}\left(\hat p + \frac{z_{\alpha/2}^2}{2n} \pm z_{\alpha/2}\sqrt{\frac{\hat p(1 - \hat p)}{n} + \frac{z_{\alpha/2}^2}{4n^2}}\right)$$

Or equivalently,

$$\frac{1}{1 + \frac{z_{\alpha/2}^2}{n}}\left(\hat p + \frac{z_{\alpha/2}^2}{2n}\right) \pm \frac{z_{\alpha/2}}{1 + \frac{z_{\alpha/2}^2}{n}}\sqrt{\frac{\hat p(1 - \hat p)}{n}+\frac{z_{\alpha/2}^2}{4n^2}}$$

The expression to the left of the "$\pm$" is the midpoint of the confidence interval and the expression to the right is the margin of error.

In [3]:
midpoint <- 1/(1+z^2/n)*(pHat + z^2/(2*n))
E <- 1/(1+z^2/n)*z*sqrt(pHat*(1 - pHat)/n + z^2/(4*n^2))
E
midpoint - E
midpoint + E

[1] 0.02105517

[1] 0.4589804

[1] 0.5010907

So we're 95% confident that the population proportion of Biden voters is $p\in(0.45898, 0.50109)$.  Pretty close to the Wald interval!